benchmark/QR_accuracy_stability_analysis.cc

```diff
@@ -15,148 +15,317 @@
 
 template <typename T>
 static void
-test_cond_helper_0(int64_t m, 
+error_check(int64_t m,
+            int64_t n,
+            int64_t k,
+            T norm_A,
+            T* A_dat,
+            T* A_1_dat,
+            T* Q_dat,
+            T* R_dat,
+            T* I_ref_dat
+            ) {
+
+    // Check orthogonality of Q
+    // Q' * Q  - I = 0
+    blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, k, k, m, 1.0, Q_dat, m, Q_dat, m, -1.0, I_ref_dat, k);
+    T norm_0 = lapack::lange(lapack::Norm::Fro, k, k, I_ref_dat, k);
```

Comment:
Use ``syrk`` and ``lansy`` for forming the Gram matrix and computing the Frobenius norm of a symmetric matrix.

I see that you're making this mistake pretty often. Please do a CTRL+F or a grep for the pattern ``blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans``. You should scrutinize any use case which looks like that and see if you're taking a Gram matrix. If you are, then you should use ``syrk`` and then copy into the other triangle _or_ you should modify downstream function calls appropriately. For an example of the latter category, you'd need to replace ``lange`` calls with ``lansy`` calls.

Reply:
Yeah, that was an issue indded. Propagated from me copying stuff from last summer. Should be all fixed now.